File: csharp/darts/Darts.cs
using System;

public static class Darts
{
    public static int Score(double x, double y)
    {
        double hypotenuse = Hypotenuse(x, y);
        if (hypotenuse > 10)
        {
            return 0;
        }

        if (hypotenuse > 5)
        {
            return 1;
        }

        if (hypotenuse > 1)
        {
            return 5;
        }

        return 10;
    }

    public static double Hypotenuse(double x, double y) => Math.Sqrt(Math.Pow(x, 2) + Math.Pow(y, 2));
}

File: dart/darts/lib/darts.dart
import 'dart:math';

class Darts {
  int score(double x, double y) {
    double distance = pythagoreanLength(x, y);
    if (distance > 10) {
      return 0;
    }
    if (distance > 5) {
      return 1;
    }
    if (distance > 1) {
      return 5;
    }
    return 10;
  }

  double pythagoreanLength(double x, double y) {
    return sqrt(pow(x, 2) + pow(y, 2));
  }
}
